src/js/modules/autocomplete.js

```diff
@@ -157,68 +175,69 @@ var chatInputKeyupFunc = function(e){
 
   if ((e.keyCode === KEYS.enter || e.keyCode === KEYS.tab) && hasItems) {
     e.preventDefault();
-    previewList.selected = $('.preview-item.selected').find('.ac-text').text();
+    previewList.selected = $(".preview-item.selected").find(".ac-text").text();
     previewList.updateChatInput();
     return false;
   }
 
   var currentText = this.value;
   var cursorPos = $(this).get(0).selectionStart;
-  
+
   var triggerMatch = currentText.match(inputRegex);
-  
+
   var previewSearchStr = "";
 
   if (triggerMatch && triggerMatch.length > 0) {
-    var matchData = handleMatch(triggerMatch, currentText, cursorPos, keyCharMin);
+    var matchData = handleMatch(
+      triggerMatch,
+      currentText,
+      cursorPos,
+      keyCharMin
+    );
     previewSearchStr = matchData.currentMatch;
     previewList.data = matchData;
   }
 
-  log("inKeyup",previewList.data);
+  log("inKeyup", previewList.data);
 
   shouldClearPreview(acPreview, previewSearchStr, currentText, keyCharMin);
-
 };
 
-var chatInputKeydownFunc = function(e){
-  var emptyPreview = document.querySelector('#autocomplete-preview').children.length <= 0;
-  var isValidKey = _.includes([
-    KEYS.tab, 
-    KEYS.enter, 
-    KEYS.up, 
-    KEYS.down,
-    KEYS.left,
-    KEYS.right
-  ], 
+var chatInputKeydownFunc = function (e) {
+  var emptyPreview = !document.querySelector("#autocomplete-preview > li");
+  var isValidKey = _.includes(
+    [KEYS.tab, KEYS.enter, KEYS.up, KEYS.down, KEYS.left, KEYS.right],
```

Comment:
Could we be sending enter twice here? Shift + Enter handled once by Queueup, and then an extra enter handled by Dub+?

Reply:
it's something in their code. I see them manually inserting the `\n` on their end